### PR TITLE
[Merged by Bors] - Remove dest-file on fail & trimming url (close #203)

### LINKED ISSUE #203: Build on windows failed ( > node 18.x.x)
"Fresh" attempt to build with `nj-cli build` gives:
```
 Compiling chrono-tz-build v0.0.3
   Compiling http_req v0.8.1
   Compiling nj-build v0.2.4 (https://github.com/infinyon/node-bindgen.git?branch=master#cf382ddf)
   Compiling chrono-tz v0.6.3
   Compiling node-bindgen v5.0.0 (https://github.com/infinyon/node-bindgen.git?branch=master#cf382ddf)
   Compiling node-bindings v0.1.0 (C:\Users\username\projects\application_name\application\apps\rustcore\rs-bindings)
error: failed to run custom build command for `node-bindings v0.1.0 (C:\Users\username\projects\application_name\application\apps\rustcore\rs-bindings)`

Caused by:
  process didn't exit successfully: `C:\Users\username\projects\application_name\application\apps\rustcore\rs-bindings\target\debug\build\node-bindings-d5696
c941734570c\build-script-build` (exit code: 101)
  --- stdout
  downloading nodejs: https://nodejs.org/dist/v18.12.1
  /win-x64/node.lib to: "C:\\Users\\username\\AppData\\Local\\Temp\\node-v18.12.1.lib"

  --- stderr
  thread 'main' panicked at 'Download node.lib file failed: Parse(StatusErr)', C:\Users\username\.cargo\git\checkouts\node-bindgen-5f69c8b3ac8cdb9e\cf
382dd\nj-build\src\lib.rs:35:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```
Looks like the script tries to download lib and fails with it. In the destination folder `C:\\Users\\username\\AppData\\Local\\Temp` file `node-v18.12.1.lib` exists but the size is 0. 
Well with next attempt to build with `nj-cli build` we have link error as result

```
error: linking with `link.exe` failed: exit code: 1136
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\link.exe" "/DEF:C:\\
windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\username\\.rustup\\toolchains\\1.61.0-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\
(...cutted...)
\libcore.natvis" "/NATVIS:C:\\Users\\username\\.rustup\\toolchains\\1.61.0-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: C:\Users\username\AppData\Local\Temp\node-v18.12.1.lib : fatal error LNK1136: invalid or corrupt file

error: could not compile `node-bindings` due to previous error
```
A way around to fix it:
- manually download `node-v18.12.1.lib` from https://nodejs.org/dist/v18.12.1/win-x64/
- copy `node-v18.12.1.lib` into `C:\\Users\\username\\AppData\\Local\\Temp`
- build

It's more or less okay for local use but makes things really complicated on github actions.

Could you please take a look at it?

notes:
* "Fresh" - with removing `Cargo.lock` file
* tried with node 18.12.0, 18.8.0 and 19.x.x
* tried to uninstall/reinstall/repair node and vstools

### PR DESCRIPTION
**Pull request by DmitryAstafyev**
- trim version of node before build target download-url
- remove dest-file in case if download fails

### CODE REVIEW

**Comment by DmitryAstafyev** [comment]
related to #203

**Comment by sehz** [comment]
CI failed due to clippy and fmt.

**Comment by DmitryAstafyev** [comment]
@sehz put your attention to the last commit. I've putted this peace of "clippy resolutions" as separated commit because it's quite important change

**Comment by DmitryAstafyev** [comment]
@sehz please let me know if multiple commits aren't okay. I will rebase it into 1 commit.

**Comment by sehz** [comment]
Thanks for your contribution.  Multiple commits are fine. bors will squash anyway

File: nj-core/src/class.rs [hunk 1]
```diff
@@ -158,7 +158,7 @@ pub trait JSClass: Sized {
         debug!("my object finalize");
         unsafe {
             let ptr: *mut JSObjectWrapper<Self> = finalize_data as *mut JSObjectWrapper<Self>;
-            Box::from_raw(ptr);
+            drop(Box::from_raw(ptr));
```
[the remark this answers is not in the document]
Comment (DmitryAstafyev) [reply nj-core/src/class.rs:161]: clippy recommendation

[the remark this answers is not in the document]
Comment (sehz) [reply nj-core/src/class.rs:161]: this changes behavior, let's suppress this

[the remark this answers is not in the document]
Comment (DmitryAstafyev) [reply nj-core/src/class.rs]: would it be better?
```
let _ = Box::from_raw(ptr);
```

[the remark this answers is not in the document]
Comment (DmitryAstafyev) [reply nj-core/src/class.rs]: or you suggest just ignore clippy warning on it?

[the remark this answers is not in the document]
Comment (sehz) [reply nj-core/src/class.rs]: both would work as long as behavior doesn't change and satisfy clippy

**Review by sehz — APPROVED** [review]
LGTM

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**